packages/core/src/tracing/spanBuffer.js

```diff
@@ -483,3 +501,28 @@ function removeSpansIfNecessary() {
     spans = spans.slice(-maxBufferedSpans);
   }
 }
+/**
+ * @param {import('../core').InstanaBaseSpan} span
+ * @returns {import('../core').InstanaBaseSpan} span
+ */
+function processSpan(span) {
+  if (!span || typeof span.n !== 'string') {
```

Comment:
suggestion: What was the reason to add this validation here? If you would like to validate the object, it belongs to the `addSpan` function or even earlier in the chain. This gives u also the chance to add a better error message.